guardian/editorial-tools-user-telemetry-service
# ci: Adds semantic release

## What does this change?

This PR attempts to add semantic release to our CI process, meaning people don't have to manually release packages to NPM. It should mean when changes are merged to main, a release will automatically be handled if one is applicable.

Its uses the semantic-release package and the conventional commits format. It should also only fire for changes to the `user-telemetry-client` subproject. 

## How to test


Do the CI processes kick off correctly for a branch? Once merged, will the trigger a deploy (we can practice this using the beta branch see [action](https://github.com/guardian/editorial-tools-user-telemetry-service/runs/5553198626?check_suite_focus=true) and [npm](https://www.npmjs.com/package/@guardian/user-telemetry-client/v/1.0.0-beta.2))? 

## How can we measure success?



Package versioning and release is a breeze!
